### PR TITLE
assorted fixes

### PR DESCRIPTION
**Pull request by xeno-by**

### CODE REVIEW

File: scalameta/trees/shared/src/main/scala/scala/meta/internal/ast/LiteralFormatter.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package scala.meta.internal.ast
+
+trait LiteralFormatter[T] {
```
Comment (olafurpg) [inline scalameta/trees/shared/src/main/scala/scala/meta/internal/ast/LiteralFormatter.scala]: Remove?

**Comment by xeno-by** [comment]
@olafurpg Any ideas about what's wrong with this build?

**Comment by olafurpg** [comment]
The CI server had some memory issues recently causing flaky builds. I hit on the same issue in the Scala.js PR, restarting the build fixed the issue.

The platform-ci is new and we are still learning how to scale it up. We probably need to monitor the server resource usage more closely. I'll discuss with @jvican about making sure we avoid more flaky builds.

**Comment by olafurpg** [comment]
Restarting the build does (fortunately) not help. I'm gonna try and lower the `-Xmx` settings https://github.com/scalameta/scalameta/pull/764

**Comment by olafurpg** [comment]
The flaky build failures are unrelated to this change so I'm gonna go ahead and merge this. Let's hope #764 fixes the flakiness.

**Comment by jvican** [comment]
I've done some changes to the server and now it should perform better. We should not experience the flaky errors. Let me know if they persist.

**Pull request merged**